### ISSUE TITLE
Complete high contrast theme

### ISSUE THREAD

**Issue by Joelkw** [sourcegraph/sourcegraph#16736]
Complete the high contrast theme that @sqs started in order to better comply with Web Content Accessibility Guidelines standards. 

WIP on [this branch](https://github.com/sourcegraph/sourcegraph/tree/sqs/hc-black-theme).

This is related to #14546 .

**Comment by Joelkw** [comment]
This is related to #14964 and #4640, but could be a potential faster fix or used to inform our work on those custom theme concepts.

**Comment by felixfbecker** [comment]
@Joelkw I disagree that it would close https://github.com/sourcegraph/sourcegraph/issues/14546. Imo the default theme should (also) pass common contrast guidelines, a high contrast theme is something for users that need _even higher_ contrast than that (and would not be expected to be used by the majority of users, because it is a trade-off).

**Comment by Joelkw** [comment]
Got it, and fair point. Updated.

**Comment by quinnkeast** [comment]
Some quick thoughts:

- I completely agree with @felixfbecker that the default theme should pass all accessibility guidelines. Creating a high-contrast theme for this purpose encodes the idea that "accessibility" is a "mode," which is against the principles of inclusive design. We must do everything possible to make the baseline themes meet accessibility standards before we invest in making new themes.
  - Now to totally mess with this statement: if this solves systematic issues we can't easily address with the current default themes, which would be better solved in the upcoming redesign... then we should get this done ASAP, while reinforcing our commitment to getting it right from the start with the redesign.
- A _higher_ contrast theme supports specific beyond-the-baseline accessibility needs. As a designer who doesn't need this level of support to meet my needs, I wouldn't be able to evaluate whether my design is actually solving these accessibility needs; the best I could do is to check contrast calculations, but those are very specific and fine-grained and can't address the bigger, holistic UI. For example: all the contrast calculations might say it works, but the hierarchy and differentiation between elements might be lost. (I'm reminded of "[Building the most inaccessible site possible with a perfect Lighthouse score](https://www.matuzo.at/blog/building-the-most-inaccessible-site-possible-with-a-perfect-lighthouse-score/).")

I think we should co-design this high-contrast theme with developers who need higher contrast to support their work. We should seek developers who explicitly have this need, and work with them to create this theme based on their feedback. (We should treat this as a consultation and compensate accordingly.)

**Comment by AlicjaSuska** [comment]
@jeanduplessis please see the above in the context of our iteration planning ☝️

**Comment by felixfbecker** [comment]
@quinnkeast 100% agree, well said!!

> if this solves systematic issues we can't easily address with the current default theme

Did you have any in mind here?

**Comment by quinnkeast** [comment]
>Did you have any in mind here?

Not directly—just very pragmatically aware that accumulated debt (design and tech) can make this much easier said than done. 😄

**Comment by jasongornall** [comment]
cc @umpox being handled by you as part of our accessiblity work. We are going to update existing styles with a feature flag. 
feel free to mention in [champions](https://sourcegraph.slack.com/archives/C038K3PR2M7) if you have any concerns
